### PR TITLE
[WIP] Adding CDI e2e negative tests

### PR DESCRIPTION
**Pull request by zvikorn**
One of the tests I created is 'import invalid credentials' test case which fails right now as the source of the disk image does not require credentials. what should I test this against? should we create another container/pod for this specific test that returns error for invalid credentials?

### CODE REVIEW

File: tests/cloner_test.go [hunk 1]
```diff
@@ -121,3 +125,32 @@ func doCloneTest(f *framework.Framework, targetNs *v1.Namespace) {
...
 	}
 }
+
+func verifyTargetContent(f *framework.Framework, namespace *v1.Namespace, pvc *v1.PersistentVolumeClaim) bool {
+	By("Verify target PVC content matches source PVC")
+	executorPod, err := utils.CreateExecutorPodWithPVC(f.K8sClient, "verify-pvc-content", namespace.Name, pvc)
+	Expect(err).ToNot(HaveOccurred())
+	err = utils.WaitTimeoutForPodReady(f.K8sClient, executorPod.Name, namespace.Name, utils.PodWaitForTime)
+	Expect(err).ToNot(HaveOccurred())
+	output := f.ExecShellInPod(executorPod.Name, namespace.Name, "cat "+testFile)
```
Comment (jeffvance) [inline tests/cloner_test.go]: Is there value in this func also returning err?

File: tests/import_test.go [hunk 2]
```diff
@@ -6,18 +6,44 @@ import (
...
+	bedEndpoint             = "http://gopats.com/who-is-the goat.iso"
+	endpoint                = "http://distro.ibiblio.org/tinycorelinux/9.x/x86/release/Core-current.iso" 
+	annServiceAccountName   = "kubernetes.io/service-account.name"
+	secretName              = "mysecret"
 	waitForCDIImportTimeout = 30 * time.Second
 )
 
 var _ = Describe(testSuiteName, func() {
-	f := framework.NewFrameworkOrDie(namespacePrefix)
+	f, err := framework.NewFramework(namespacePrefix, framework.Config{})
```
Comment (jeffvance) [inline tests/import_test.go]: Consider calling the new [`NewFrameworkOrDie`](https://github.com/kubevirt/containerized-data-importer/blob/master/tests/framework/framework.go#L101). You do not need to check err and you do not need to pass in Config.

**Comment by jeffvance** [comment]
looks good to me.
Needs rebase but should be ready to merge.

**Comment by awels** [comment]
@zvikorn did you close this on purpose?

File: tests/cloner_test.go [hunk 3]
```diff
@@ -121,3 +126,32 @@ func doCloneTest(f *framework.Framework, targetNs *v1.Namespace) {
...
+func cloneOfAnnoExistenceTest(f *framework.Framework, targetNs *v1.Namespace) {
+	// Create targetPvc
+	By("Creating target pvc")
+	targetPvc, err := utils.CreatePVCFromDefinition(f.K8sClient, targetNs.Name, utils.NewPVCDefinition(
+		"target-pvc",
+		"1G",
+		map[string]string{controller.AnnCloneRequest: f.Namespace.Name + "/" + sourcePVCName, controller.AnnCloneOf: "true"},
+		nil))
+	Expect(err).ToNot(HaveOccurred())
+	err = utils.WaitForPersistentVolumeClaimPhase(f.K8sClient, targetNs.Name, v1.ClaimBound, targetPvc.Name)
```
Comment (awels) [inline tests/cloner_test.go]: I don't think you can wait for ClaimBound, as with local volume that will not happen unless a pod attaches the PVC to itself. Which won't happen.

**Comment by zvikorn** [comment]
Yes. I’m going to open a new one soon. 
I saw your comment though. Thanks!

File: tests/cloner_test.go (hunk 3 above)
Comment (awels) [reply tests/cloner_test.go]: So just merged a patch that should make it easier to inspect the controller log to see if it is skipping the PVC you are interested in.

**Pull request closed without merging**